modules/tensor_mechanics/src/materials/FiniteStrainCrystalPlasticity.C

```diff
@@ -729,17 +735,20 @@ FiniteStrainCrystalPlasticity::solveStress()
   unsigned int iter = 0;
   RankTwoTensor resid, dpk2;
   RankFourTensor jac;
+  Real rnorm, rnorm0, rnorm_prev;
 
   calc_resid_jacob(resid, jac); // Calculate stress residual
   if (_err_tol)
   {
-    mooseWarning("FiniteStrainCrystalPLasticity: Slip increment exceeds tolerance - Element number " << _current_elem->id() << " Gauss point = " << _qp);
+    //    mooseWarning("FiniteStrainCrystalPLasticity: Slip increment exceeds tolerance - Element number " << _current_elem->id() << " Gauss point = " << _qp);//Use for debugging
     return;
   }
 
-  Real rnorm = resid.L2norm();
+  rnorm = resid.L2norm();
+  rnorm0 = rnorm;
 
-  while (rnorm > _rtol && iter <  _maxiter) // Check for stress residual tolerance
+  while (rnorm > _rtol * rnorm0 && rnorm0 > _abs_tol && iter <  _maxiter) // Check for stress residual tolerance
+  //  while (rnorm > _rtol && iter <  _maxiter) // Check for stress residual tolerance
```

Comment:
Let's not keep any commented out code in here. It is stored in the git history anywas